### PR TITLE
allow nan as no data and check value for validity

### PR DESCRIPTION
**Pull request by michaelmattig**

### CODE REVIEW

**Comment by michaelmattig** [comment]
[Expression: unable to set NaN as nodata for float](https://trello.com/c/E6kx4TpN/35-expression-unable-to-set-nan-as-nodata-for-float)

File: projects/wave-core-new/src/lib/operators/dialogs/expression-operator/expression-operator.component.html [hunk 1]
```diff
@@ -41,7 +41,7 @@
 
         <mat-form-field>
             <mat-label>No-Data Value</mat-label>
-            <input matInput type="number" formControlName="noDataValue" />
+            <input matInput type="text" formControlName="noDataValue" />
```
Comment (ChristianBeilschmidt) [inline projects/wave-core-new/src/lib/operators/dialogs/expression-operator/expression-operator.component.html:44]: Why text?

File: projects/wave-core-new/src/lib/operators/dialogs/expression-operator/expression-operator.component.ts [hunk 2]
```diff
@@ -152,6 +153,43 @@ export class ExpressionOperatorComponent implements AfterViewInit, OnDestroy {
         // this.unitSubscription.unsubscribe();
     }
 
+    validateNoData(control: AbstractControl): ValidationErrors | null {
+        if (!isNaN(parseFloat(control.value)) || control.value.toLowerCase() === 'nan') {
```
Comment (ChristianBeilschmidt) [inline projects/wave-core-new/src/lib/operators/dialogs/expression-operator/expression-operator.component.ts:157]: You can express no data as a number, but just not with JSON.
IMHO you should work with number here and as potential string for submitting it to our server.

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: projects/wave-core-new/src/lib/operators/dialogs/expression-operator/expression-operator.component.html (hunk 1 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/operators/dialogs/expression-operator/expression-operator.component.html:44]: s.t. users can set NAN as the no-data value. That was the whole point of the task... Now they can either write a number or "nan" in order to create the layer

File: projects/wave-core-new/src/lib/operators/dialogs/expression-operator/expression-operator.component.ts (hunk 2 above)
Comment (michaelmattig) [reply projects/wave-core-new/src/lib/operators/dialogs/expression-operator/expression-operator.component.ts:157]: users have to be able to set NAN as the selected no data value

File: projects/wave-core-new/src/lib/operators/dialogs/expression-operator/expression-operator.component.html (hunk 1 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/operators/dialogs/expression-operator/expression-operator.component.html:44]: Hm, I see. We could allow NAN only for floats and we should then present a checkbox for NAN.
This box could disable the number field then.
What do you think?

Comment (michaelmattig) [reply projects/wave-core-new/src/lib/operators/dialogs/expression-operator/expression-operator.component.html:44]: that would also be possible. however the way I did it is also fine imho.

**Pull request merged**